packages/flutter/lib/src/services/raw_keyboard.dart

```diff
@@ -6,9 +6,79 @@ import 'dart:async';
 
 import 'package:flutter/foundation.dart';
 
+import 'raw_keyboard_android.dart';
+import 'raw_keyboard_fuschia.dart';
 import 'system_channels.dart';
 
-/// Base class for platform specific key event data.
+/// An enum describing the side of the keyboard that a key is on, to allow
+/// discrimination between which key is pressed (e.g. the left or right SHIFT
+/// key).
+///
+/// See also:
+///
+///  * [RawKeyEventData.isModifierPressed], which accepts this enum as an
+///    argument.
+enum KeyboardSide {
+  /// Matches either the left or right version of the key.
+  any,
+
+  /// Matches the left version of the key.
+  left,
+
+  /// Matches the right version of the key.
+  right,
+
+  /// Matches the left and right version of the key pressed simultaneously.
+  both,
+}
+
+/// An enum describing the type of modifier key that is being pressed.
+///
+/// See also:
+///
+///  * [RawKeyEventData.isModifierPressed], which accepts this enum as an
+///    argument.
+enum ModifierKey {
+  /// The CTRL modifier key. Typically, there are two of these.
```

Comment:
nit for many doc comments in this file: The style guide is to use only one sentence in the first paragraph.